plugins/ikfastsolvers/ikfastmodule.cpp

```diff
@@ -1145,29 +1158,40 @@ class IkFastModule : public ModuleBase
                     }
                 }
 
-                if( !pmanip->GetIkSolver()->GetFreeParameters(vfreeparameters_real) ) {
+                if( !piksolver->GetFreeParameters(vfreeparameters_real) ) {
                     RAVELOG_WARN("failed to get freeparameters");
                 }
                 if( bTestSelfCollision && robot->CheckSelfCollision()) {
                     RAVELOG_VERBOSE("robot in self-collision\n");
                     continue;
                 }
 
+                if( !piksolver->GetFreeParameters(vfreeparameters) ) {
+                    RAVELOG_WARN("failed to get freeparameters");
+                }
+                vfreeparameters_out.resize(vfreeparameters.size());
+                
                 // have to start at a random config
                 while(1) {
-                    for(int j = 0; j < (int)vrand.size(); j++) {
+                    for(int j = 0; j < (int) narminds; j++) {
                         vrand[j] = vlowerlimit[j] + (vupperlimit[j]-vlowerlimit[j])*RaveRandomFloat();
                     }
+                    for(int k = 0; k < (int) nfreeparams; k++) {
```

Comment:
whole point of the random test is to not rely on the randomly chosen freevalue...

Reply:
add an extra test that computes with free joint values known; keep the original test that computes with free joint values unknown

Reply:
can pass "--freeindices" on command line to "openrave.py --database inversekinematics"; print out five percentages from DebugIK test in testik from inversekinematics.py